### PR TITLE
Add [edit] links to all pages

### PR DESCRIPTION
**Pull request by NeQuissimus**
Adds a link to each page labelled "[edit]", which redirects to an edit on GitHub.

@fommil with the great ideas again :+1:

### CODE REVIEW

**Comment by fommil** [comment]
Fommil stealing all his ideas from @dwijnand and passing them off as his own... you mean :wink:

**Comment by hrhino** [comment]
:+1: to every page by default

**Comment by NeQuissimus** [comment]
Let me see if that is possible... There is no such thing as a header but maybe I can create a custom layout

**Comment by dwijnand** [comment]
> Fommil stealing all his ideas from @dwijnand and passing them off as his own... you mean 😉

I stole from you (ENSIME). And your stuff in ENSIME is @rorygraves "stealing" from tomjohnson1492/documentation-theme-jekyll 🙂

https://github.com/ensime/ensime.github.io/commit/01ef378ed8feedcd3ec1c4f44f0f43280fdb919c#diff-831ca5472e031dc0bb63f628a8d8b72eR18

**Comment by NeQuissimus** [comment]
This is turning into a rabbit hole of criminals xD

File: site/src/main/resources/microsite/layouts/scalaz.html [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                  </li>
+                </ul>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div id="content" data-github-owner="scalaz" data-github-repo="scalaz">
+          <div class="content-wrapper">
+            <section>
+              <a href="https://github.com/scalaz/scalaz/edit/series/7.3.x/site/src/main/tut/{{ page.path }}" style="float:right;">[edit]</a>
```
Comment (NeQuissimus) [inline site/src/main/resources/microsite/layouts/scalaz.html]: BAM!

**Comment by NeQuissimus** [comment]
This wasn't so bad...

**Comment by xuwei-k** [comment]
https://github.com/scalaz/scalaz/pull/1921

**Pull request merged**